### PR TITLE
Settings page

### PR DESCRIPTION
**Pull request by tikitko**

### CODE REVIEW

File: blog-generic/src/entities/common_minimal_author.rs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    #[validate(length(min = 2, max = 50), non_control_character)]
+    pub last_name: Option<String>,
+    #[validate(length(max = 255), url)]
+    pub image_url: Option<String>,
+}
+
+fn validate_slug(slug: &String) -> Result<(), ValidationError> {
+    let chars_count = slug.chars().count();
+    if !(5 <= chars_count && chars_count <= 30) {
+        return Err(ValidationError::new("is_not_in_renge_5_30"));
```
Comment (YoshkiMatryoshki) [inline blog-generic/src/entities/common_minimal_author.rs:20]: is_not_in_r**A**nge_5_30

File: blog-generic/src/entities/common_minimal_author.rs [hunk 2]
```diff
@@ -0,0 +1,31 @@
+use serde::{Deserialize, Serialize};
+use validator::{Validate, ValidationError};
+
+#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize, Validate)]
+#[serde(rename_all = "camelCase")]
+pub struct CommonMinimalAuthor {
```
Comment (YoshkiMatryoshki) [inline blog-generic/src/entities/common_minimal_author.rs:6]: AuthorProfile, ProfileInfo maybe?

File: blog-server-api/src/endpoints/author_override_social_data/handler.rs [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    let author = auth_author_future.await.map_err(|e| Unauthorized {
+        reason: e.to_string(),
+    })?;
+
+    if author.base.blocked == 1 {
+        return Err(Forbidden);
+    }
+
+    author_service
+        .set_author_override_social_data_by_id(&author.id, &override_social_data)
```
Comment (YoshkiMatryoshki) [inline blog-server-api/src/endpoints/author_override_social_data/handler.rs:43]: we can simplify name with set_override_social_data

Comment (tikitko) [reply blog-server-api/src/endpoints/author_override_social_data/handler.rs:43]: trying to keep only naming style in service scope

File: blog-generic/src/entities/common_minimal_author.rs (hunk 2 above)
Comment (tikitko) [reply blog-generic/src/entities/common_minimal_author.rs:6]: trying to keep similar naming style with other structs

**Review by YoshkiMatryoshki — APPROVED** [review]

**Pull request merged**